PlayTools/PlaySettings.swift

```diff
@@ -1,188 +1,103 @@
-        if let key = settings[PlaySettings.ipadModelKey] as? NSString {
-            switch key {
-            case "iPad6,7":
-                return "J98aAP"
-            case "iPad8,6":
-                return "J320xAP"
-            case "iPad13,8":
-                return "J522AP"
-            default:
-                return "J320xAP"
-            }
-        }
-        return "J320xAP"
-    }()
-
-    static var isGame: Bool {
-        if let info = Bundle.main.infoDictionary?.description {
-            for keyword in PlaySettings.keywords {
-                if info.contains(keyword) && !info.contains("xbox") {
-                    return true
-                }
-            }
-        }
-        return false
-    }
+    @objc lazy var adaptiveDisplay = isGame
 
-    lazy var settings: [String: Any] = {
-        if let prefs = allPrefs[Bundle.main.bundleIdentifier!] as? [String: Any] {
-            return prefs
-        }
-        return [PlaySettings.adaptiveDisplayKey: PlaySettings.isGame, PlaySettings.keymappingKey: PlaySettings.isGame]
-    }()
+    @objc lazy var deviceModel = settingsData.iosDeviceModel as NSString
 
-    lazy var allPrefs: [String: Any] = {
-        do {
-            if let all = try [String: Any].read(PlaySettings.settingsUrl) {
-                return all
-            }
-        } catch {
-            print("failed to load settings: \(error)")
+    @objc lazy var oemID: NSString = {
+        switch settingsData.iosDeviceModel {
+        case "iPad6,7":
+            return "J98aAP"
+        case "iPad8,6":
+            return "J320xAP"
+        case "iPad13,8":
+            return "J522AP"
+        default:
+            return "J320xAP"
         }
-        return [:]
     }()
 
-    public func clearLegacy() {
-        UserDefaults.standard.removeObject(forKey: "layout")
-        UserDefaults.standard.removeObject(forKey: "pclayout")
-        UserDefaults.standard.removeObject(forKey: "playcover.macro")
-        UserDefaults.standard.removeObject(forKey: PlaySettings.sensivityKey)
-        UserDefaults.standard.removeObject(forKey: PlaySettings.refreshRateKey)
-        UserDefaults.standard.removeObject(forKey: PlaySettings.keymappingKey)
-        UserDefaults.standard.removeObject(forKey: PlaySettings.adaptiveDisplayKey)
-        UserDefaults.standard.removeObject(forKey: PlaySettings.gameWindowSizeWidth)
-        UserDefaults.standard.removeObject(forKey: PlaySettings.gameWindowSizeHeight)
-        UserDefaults.standard.removeObject(forKey: PlaySettings.enableWindowAutoSize)
-    }
+    @objc lazy var refreshRate = settingsData.refreshRate
 
-    public static let settingsUrl = URL(fileURLWithPath: "/Users/\(NSUserName())/Library/Preferences/playcover.plist")
+}
 
-    private static var keywords = ["game", "unity", "metal", "netflix", "opengl", "minecraft", "mihoyo", "disney"]
+struct AppSettingsData: Codable {
+    var keymapping = true
+    var mouseMapping = true
+    var keymap: [[CGFloat]] = []
```

Comment:
Keymapping has been moved to a separate file, and its structure has changed.

Reply:
https://github.com/PlayCover/PlayCover/blob/2.0-UI/PlayCover/Model/Keymapping.swift